server/utils.go

```diff
@@ -154,7 +154,12 @@ func isContextError(err error) bool {
 func (s *Server) getResourceOrWait(ctx context.Context, name, resourceType string) (string, error) {
 	// In 99% of cases, we shouldn't hit this timeout. Instead, the context should be cancelled.
 	// This is really to catch an unlikely case where the kubelet doesn't cancel the context.
-	const resourceCreationWaitTime = time.Minute * 6
+	// Adding on top of the specified deadline ensures this deadline will be respected, regardless of
+	// how Kubelet's runtime-request-timeout changes.
+	resourceCreationWaitTime := time.Minute * 4
```

Comment:
This was set to 6 prior, should it still be?

```suggestion
	resourceCreationWaitTime := time.Minute * 6
```

Reply:
I've decided that 6 along with the other timeout (2 minutes usually) is excessive, especially because the runtime timeout is typically only increased afaik

Reply:
@fgiudici that's fine by me.  I was asking primarily to make sure it wasn't an inadvertent change.